libraries/c_sdk/standard/https/src/iot_https_client.c

```diff
@@ -2464,6 +2344,56 @@ IotHttpsReturnCode_t IotHttpsClient_Init( void )
     _httpParserSettings.on_body = _httpParserOnBodyCallback;
     _httpParserSettings.on_message_complete = _httpParserOnMessageCompleteCallback;
 
+    /* Allocate the dispatch queue. */
+    #if IOT_HTTPS_DISPATCH_USE_STATIC_MEMORY == 1
+        dispatchQueue = xQueueCreateStatic( IOT_HTTPS_DISPATCH_QUEUE_SIZE,
+                                            sizeof( _httpsRequest_t * ),
+                                            dispatchQueueStorageBuffer,
+                                            &dispatchQueueBuffer );
+    #else
+        dispatchQueue = xQueueCreate( IOT_HTTPS_DISPATCH_QUEUE_SIZE, sizeof( _httpsRequest_t * ) );
+    #endif
+
+    if( dispatchQueue == NULL )
+    {
+        /* Queue was not created and must not be used. */
+        IotLogError( "Failed to allocate resources for dispatch queue.", status );
+        HTTPS_SET_AND_GOTO_CLEANUP( IOT_HTTPS_INTERNAL_ERROR );
+    }
+
+    /* Start tasks that send requests from the dispatch queue. */
+    for( dispatchTaskIndex = 0; dispatchTaskIndex < IOT_HTTPS_DISPATCH_TASK_COUNT; ++dispatchTaskIndex )
+    {
+        #if IOT_HTTPS_DISPATCH_USE_STATIC_MEMORY == 1
+            httpsDispatchTask[ dispatchTaskIndex ] = xTaskCreateStatic( _dispatchTaskRoutine,
+                                                                        "iot_thread",
+                                                                        IOT_HTTPS_DISPATCH_TASK_STACK_SIZE,
+                                                                        NULL,
+                                                                        IOT_HTTPS_DISPATCH_TASK_PRIORITY,
+                                                                        dispatchTaskStack,
+                                                                        &dispatchTaskBuffer );
+
+            if( httpsDispatchTask[ dispatchTaskIndex ] == NULL )
+            {
+                IotLogError( "Failed to allocate static memory for request task.", status );
+                HTTPS_SET_AND_GOTO_CLEANUP( IOT_HTTPS_INTERNAL_ERROR );
+            }
+        #else /* ifdef IOT_HTTPS_DISPATCH_USE_STATIC_MEMORY */
+            taskCreationResult = xTaskCreate( _dispatchTaskRoutine,
```

Comment:
How will you clean up these non-static tasks when the HTTP library is done?

Reply:
> NOTE: The idle task is responsible for freeing the RTOS kernel allocated memory from tasks that have been deleted. It is therefore important that the idle task is not starved of microcontroller processing time if your application makes any calls to vTaskDelete (). Memory allocated by the task code is not automatically freed, and should be freed before the task is deleted.

From my understanding of this, once we call `xTaskDelete`, then the idle task will be responsible for clean up for a non-static task.

Reply:
Excuse me I missed the httpsDispatchTask[ dispatchTaskIndex ] that stores the task in the parameter. Thanks.